File: godtools-sync/src/main/java/org/cru/godtools/sync/job/SyncFollowupJob.java
package org.cru.godtools.sync.job;

import android.support.annotation.NonNull;

import org.cru.godtools.sync.FollowupSyncTasks;

import java.io.IOException;

public final class SyncFollowupJob extends BaseSyncJob {
    static final String TAG = "sync_followup";

    @NonNull
    @Override
    protected Result onRunJob(final Params params) {
        try {
            new FollowupSyncTasks(getContext()).syncFollowups();
        } catch (IOException e) {
            return Result.RESCHEDULE;
        }
        return Result.SUCCESS;
    }

    public static void scheduleFollowupJob() {
        scheduleSyncJob(TAG);
    }
}

File: library/sync/src/main/java/org/cru/godtools/sync/job/SyncFollowupJob.kt
package org.cru.godtools.sync.job

import com.evernote.android.job.Job
import org.cru.godtools.sync.task.FollowupSyncTasks
import java.io.IOException

internal const val FOLLOWUP_JOB_TAG = "sync_followup"

fun scheduleFollowupJob() {
    scheduleSyncJob(FOLLOWUP_JOB_TAG)
}

internal class SyncFollowupJob : BaseSyncJob() {
    override fun onRunJob(params: Job.Params): Job.Result {
        return try {
            FollowupSyncTasks(context).syncFollowups()
            Job.Result.SUCCESS
        } catch (e: IOException) {
            Job.Result.RESCHEDULE
        }
    }
}
